pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -102,103 +102,98 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 	err = r.client.List(context.TODO(), &opts, clusterRoleList)
 	if err != nil {
 		reqLogger.Error(err, "Failed to get clusterRoleList")
 		return reconcile.Result{}, err
 	}
 
-	// if crClusterRoleNameList returns list of clusterRoleNames
-	crClusterRoleNameList := populateCrClusterRoleNames(instance, clusterRoleList)
-	for _, crClusterRoleName := range crClusterRoleNameList {
-
-		var clusterRoleNames []string
-		clusterRoleNames = append(clusterRoleNames, crClusterRoleName)
-
-		// helper func to update the condition of the SubjectPermission object
-		instance := controllerutil.UpdateCondition(instance, crClusterRoleName+" for clusterPermission does not exist", clusterRoleNames, true, "Failed")
-		err = r.client.Status().Update(context.TODO(), instance)
-		if err != nil {
-			reqLogger.Error(err, "Failed to update condition.")
-			return reconcile.Result{}, err
-		}
-	}
-
 	// get a list of clusterRoleBinding from k8s cluster list
 	clusterRoleBindingList := &v1.ClusterRoleBindingList{}
-	opts = client.ListOptions{Namespace: request.Namespace}
+	opts = client.ListOptions{}
 	err = r.client.List(context.TODO(), &opts, clusterRoleBindingList)
 	if err != nil {
 		reqLogger.Error(err, "Failed to get clusterRoleBindingList")
 		return reconcile.Result{}, err
 	}
+	//reqLogger.Info(fmt.Sprintf("clusterRoleList on cluster is: %+v", clusterRoleList))
 
-	// build a clusterRoleBindingNameList which consists of clusterRoleName-subjectName
-	crClusterRoleBindingNameList := buildClusterRoleBindingCRList(instance)
-
-	// check ClusterRoleBindingName
-	populateCrClusterRoleBindingNameList := populateClusterRoleBindingNames(crClusterRoleBindingNameList, clusterRoleBindingList)
-
-	// loop through crClusterRoleBindingNameList
-	for _, clusterRoleBindingName := range populateCrClusterRoleBindingNameList {
+	if len(instance.Spec.ClusterPermissions) != 0 {
+		// if crClusterRoleNameList returns list of clusterRoleNames
+		crClusterRoleNameList := populateCrClusterRoleNames(instance, clusterRoleList)
+		for _, crClusterRoleName := range crClusterRoleNameList {
 
-		// get the clusterRoleName by spliting the clusterRoleBindng name
-		clusterRBName := strings.Split(clusterRoleBindingName, "-")
-		clusterRoleName := clusterRBName[0]
-		subjectName := clusterRBName[1]
-
-		// create a new clusterRoleBinding on cluster
-		newCRB := newClusterRoleBinding(clusterRoleName, subjectName, instance.Spec.SubjectKind)
-		err := r.client.Create(context.TODO(), newCRB)
-		if err != nil {
 			var clusterRoleNames []string
-			clusterRoleNames = append(clusterRoleNames, clusterRoleName)
+			clusterRoleNames = append(clusterRoleNames, crClusterRoleName)
 
-			// update the condition if creation of a ClusterRoleBinding has failed
-			instance := controllerutil.UpdateCondition(instance, "Unable to create ClusterRoleBinding: "+err.Error(), clusterRoleNames, true, managedv1alpha1.SubjectPermissionFailed)
+			// helper func to update the condition of the SubjectPermission object
```

Comment:
Is this a TODO?

Reply:
it is not. changing comment to better reflect logic